src/core/atomic.d

```diff
@@ -157,6 +157,38 @@ version( CoreDdoc )
 }
 else version( AsmX86_32 )
 {
+    // Uses specialized asm for fast fetch and add operations
+    private HeadUnshared!(T) atomicFetchAdd(T)( ref shared T val, size_t mod ) pure nothrow @nogc
```

Comment:
Apparently (see https://github.com/D-Programming-Language/druntime/pull/1208#issuecomment-91189879), we should rename this to `atomicAddFetch`.